pennylane/tape/qscript.py

```diff
@@ -161,37 +163,44 @@ class QuantumScript:
 
     """
 
+    name: Optional[str] = static_field(default=None)
+    
+    _prep: List[Operator] = field(default_factory=list)
+    _ops: List[Operator] = field(default_factory=list)
+    _measurements : List[MeasurementProcess] = field(default_factory=list)
```

Comment:
I particularly like how with this implementation it is now clear what the "data" contained in a `quantum script` is, as it's only those three fields!

Everything else is a property computed starting from them, which could also be cached (`@property` -> `@property_cached`)

I would like some input from @albi3ro if it's along the lines of what you think is a good idea